File: benchmarks/julia/src/parse_integers.jl
using BenchmarkTools
using Test

include("utils.jl")

function parse_int(n)
    s = string(n, base = 16)
    return UInt32(parse(Int64, s, base = 16))
end

function parse_integers(t)
    for _ in 1:t
        n = rand(UInt32)
        parse_int(n)
    end
end

# Test output
@test all(parse_int(n) == n for n in 1:1000)

# Run benchmark
write_benchmark_result("parse_integers", @benchmark parse_integers(1000))

File: benchmarks/javascript/src/parse_integers.js
import { strict as assert } from "node:assert";
import { benchmark, writeBenchmarkResult, range } from "./utils.js";

const twoToThePowerOfThirtyTwoMinusOne = 4294967295;

const generateUniformRandomInteger = () =>
  Math.floor(twoToThePowerOfThirtyTwoMinusOne * Math.random());

const parseInteger = (n) => {
  let s = n.toString(16);
  return parseInt(s, 16);
};

const parseIntegers = (t) => {
  for (let i = 0; i < t; i++) {
    let n = generateUniformRandomInteger();
    parseInteger(n);
  }
};

// Test output
assert(range(1000).every((n) => parseInteger(n) == n));

// Run benchmark
let times = benchmark(parseIntegers, [10_000], 1000, 100);
writeBenchmarkResult("parse_integers", times);
